articles/logging-with-winston/index.md

```diff
@@ -0,0 +1,550 @@
+logger.warn("Hello, Winston logger, the first warning!");
+logger.warn("Hello, Winston logger, the second warning!");
+logger.error("Hello, Winston logger, the second error!");
+logger.info("Hello, Winston logger, some info!");
+logger.debug("Hello, Winston logger, a debug!");
+```
+
+This specifies that:
+
+- Only the log that falls under level `error` will be recorded in the `example.log` file.
+- Any `warn` log will be shown in the console output.
+
+With Winston, you can specify the default format to save your logs.
+
+For example, let's say we want to log in JSON format we can specify that with `Winston.format`, and log instances will be saved in JSON format.
+
+```js
+format: winston.format(
+)
+```
+
+The format takes additional log form properties such as
+
+- `ms()` - time in milliseconds since the last log was recorded.
+- `label()` - adds a label to the logged message.
+- `timestamp()` - the time stamp the log message was received.
+- `splat()` - provides string interpolation.
+
+To apply this in your logs, you need to use `format.combine`, as shown in the example below.
+
+```js
+const logConfiguration = {
+    transports: [
+        new winston.transports.Console()
+    ],
+    format: winston.format.combine(
+        winston.format.label({
+            label: `Label🏷️`
+        }),
+        winston.format.timestamp({
+           format: 'MMM-DD-YYYY HH:mm:ss'
+       }),
+        winston.format.printf(info => `${info.level}: ${info.label}: ${[info.timestamp]}: ${info.message}`),
+    )
+};
+```
+Output:
+
+```js
+info: Label🏷️: Nov-18-2020 08:10:44: Hello, Winston logger, some info!
+```
+
+### Configuring Winston With a Server
+Let’s create a simple Express server that we can do some logging using Winston. This will be a small project to get you on your feet using Winston to record logs from server requests and responses.
+
+Go ahead and install the [Express](/engineering-education/express/) library with `npm install express`.
+
+This is the structure of our small project:
+
+![Project Structure](/engineering-education/logging-with-winston/project-structure.png)
+
+- 📂Logs - will save the log files generated by the Winston file transport.
+- `app.js`🗄️ - will be our server app.
+- 📂Utils - will hold Winston `logger.js`🗄️ where we will add configurations such as Winston transport and formating.
+
+logger.js:
+```js
+const { createLogger, format, transports } = require('winston');
+require('winston-mongodb');
+
+module.exports = createLogger({
+transports:
+    new transports.File({
+    filename: './logs/example.log',
+    format:format.combine(
+        format.timestamp({format: 'MMM-DD-YYYY HH:mm:ss'}),
+        format.align(),
+        format.printf(info => `${info.level}: ${[info.timestamp]}: ${info.message}`),
+    )}),
+});
+```
+
+Will do some logging from an Express server that listens to port 3000.
+
+app.js:
+```js
+const express = require('express');
+
+// Require logger.js
+const logger = require('./utils/log');
```

Comment:
Created logger.js in utils folder but received Error: Cannot find module './utils/log' when running app.js

Reply:
@louisefindlay23 Should be './utils/logger' (pointing to the Created logger.js in utils folder ) Fixed that on [bc2188e](https://github.com/section-io/engineering-education/pull/632/commits/bc2188eb676043f61b524ab5137cad95fee72571)